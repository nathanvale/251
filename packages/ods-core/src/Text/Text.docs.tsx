import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Box } from "../Box/Box";
import { Stack } from "../Stack/Stack";
import { Text, TextProps } from "./Text";

export const docs: ComponentDocs<TextProps> = {
  category: "Content",
  componentName: "Text",
  description: "",
  propDescriptions: {
    variant: "The font style of the text.",
    tone:
      "Tones represent colours. The usage of colour in our design system is designed to have a strong correlation with the tone of voice being used.",
    truncate:
      "This will add an ellipsis to the end of the text if the text drops onto more than one line.",
    weight: "The font weight of the text.",
    inline:
      "Setting text to inline prevents the text node from take its parents full width.",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Text>The quick brown fox jumps over the lazy dog.</Text>,
    },
    additional: [
      {
        label: "Body Small Text",
        Code: () => (
          <Text variant="body-small">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Subtitle Text",
        Code: () => (
          <Text variant="subtitle">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Subtitle Small Text",
        Code: () => (
          <Text variant="subtitle-small">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Caption Text",
        Code: () => (
          <Text variant="caption">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Overline Text",
        Code: () => (
          <Text variant="overline-text">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Tone (positive)",
        Code: () => (
          <Text tone="positive">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Tone (critical)",
        Code: () => (
          <Text tone="critical">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Weight (medium)",
        Code: () => (
          <Text weight="medium">
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Truncating long text",
        Code: () => (
          <Box style={{ width: 90 }}>
            <Text truncate>The quick brown fox jumps over the lazy dog.</Text>
          </Box>
        ),
      },
      {
        label: "Text Alignment",
        Code: () => (
          <Stack space="medium">
            <Text>The quick brown fox jumps over the lazy dog.</Text>
            <Text align="center">
              The quick brown fox jumps over the lazy dog.
            </Text>
            <Text align="right">
              The quick brown fox jumps over the lazy dog.
            </Text>
          </Stack>
        ),
      },
      {
        label: "Text Alignment (responsive)",
        Code: () => (
          <Text align={["left", "center"]}>
            The quick brown fox jumps over the lazy dog.
          </Text>
        ),
      },
      {
        label: "Inline Text",
        Code: () => (
          <Text>
            The quick brown fox{" "}
            <Text inline tone="positive">
              jumps over
            </Text>{" "}
            the lazy dog.
          </Text>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Body Text",
      Code: () => <Text>Body text</Text>,
    },
    {
      label: "Tone (positive)",
      Code: () => <Text tone="positive">Positive text</Text>,
    },
    {
      label: "Tone (critical)",
      Code: () => <Text tone="critical">Critical text</Text>,
    },
    {
      label: "Weight (medium)",
      Code: () => <Text weight="medium">Medium text</Text>,
    },
  ],
};
