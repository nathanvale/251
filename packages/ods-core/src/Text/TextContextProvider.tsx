import * as React from "react";

import {
  TextToneVariants,
  TextVariants,
  TypographyWeightVariants,
} from "@origin-digital/ods-types";

export const TEXT_CONTEXT_DEFAULTS = {
  weight: "regular",
  variant: "body",
} as const;

export interface UseTextProps {
  variant: TextVariants;
  weight: Exclude<TypographyWeightVariants, "bold">;
  tone?: TextToneVariants;
  inline?: boolean;
}

export const TextContext = React.createContext<UseTextProps | false>(false);

export function mergeToContext(
  curContext: UseTextProps | false,
  props: Partial<UseTextProps>,
  defaults: UseTextProps
) {
  let mergedContext;
  if (curContext) {
    mergedContext = {
      tone: props.tone ?? (curContext && curContext.tone) ?? defaults?.tone,
      variant:
        props.variant ?? (curContext && curContext.variant) ?? defaults.variant,
      weight:
        props.weight ?? (curContext && curContext.weight) ?? defaults.weight,
      inline:
        props.inline ?? (curContext && curContext.inline) ?? defaults.inline,
    };
  } else {
    mergedContext = {
      tone: props.tone ?? defaults?.tone,
      variant: props.variant ?? defaults.variant,
      weight: props.weight ?? defaults.weight,
      inline: props.inline ?? defaults.inline,
    };
  }

  return mergedContext;
}

export interface TextContextProviderProps extends Partial<UseTextProps> {
  children?: React.ReactNode;
  defaults?: UseTextProps;
}

export function TextContextProvider({
  variant,
  weight,
  tone,
  inline,
  children,
  defaults = TEXT_CONTEXT_DEFAULTS,
}: TextContextProviderProps) {
  const curContext = React.useContext(TextContext);

  const textContextValue = React.useMemo(() => {
    return mergeToContext(
      curContext,
      {
        tone,
        variant,
        weight,
        inline,
      },
      defaults
    );
  }, [curContext, defaults, tone, variant, weight, inline]);

  return (
    <TextContext.Provider value={textContextValue}>
      {children}
    </TextContext.Provider>
  );
}
