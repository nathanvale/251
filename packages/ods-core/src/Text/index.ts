export { Text } from "./Text";
export { TextContextProvider, TextContext } from "./TextContextProvider";
export { useTextStyles, useTextStylesFromContext } from "./useTextStyles";

export type TextProps = import("./Text").TextProps;
export type UseTextProps = import("./TextContextProvider").UseTextProps;
export type TextContextProviderProps = import("./TextContextProvider").TextContextProviderProps;
