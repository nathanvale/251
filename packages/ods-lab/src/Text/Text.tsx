import React, { ReactNode, ReactType } from "react";
import styled from "styled-components";
import { Box, BoxProps } from "@origin-digital/ods-core";
import { Color } from "@origin-digital/ods-types";
import { style } from "styled-system";

import { basekick } from "../_private/hooks/typography/basekick";

export const textColor = style({
  prop: "color",
  key: "colors",
});

/* import {useText} from './_private/hooks/typography/use-text'; */

export interface TextProps {
  children?: ReactNode;
  component?: ReactType;
  size?:
    | "xxxsmall"
    | "xxsmall"
    | "xsmall"
    | "small"
    | "medium"
    | "large"
    | "xlarge"
    | "xxlarge"
    | "xxxlarge";
  color?: keyof Color;
  weight?: "regular" | "medium";
  baseline?: boolean;
  className?: string;
  align?: BoxProps["textAlign"];
}

export const Text = ({
  children,
  component = "span",
  size = "xxsmall",
  color = "grey",
  weight = "regular",
  align,

  className,
}: TextProps) => {
  const StyledBox = styled(Box)`
    ${p => {
      return basekick({
        baseFontSize: 1,
        typeSizeModifier: p.theme.typography.text[size].size,
        typeRowSpan: p.theme.typography.text[size].rows,
        descenderHeightScale: p.theme.typography.descenderHeightScale,
        capHeight: p.theme.typography.capHeightScale,
        gridRowHeight: p.theme.typography.gridRow,
      });
    }}
    font-family: gordita, sans-serif;
    font-weight: ${p => p.theme.typography.weight[weight]};
    ${textColor}
  `;
  return (
    <StyledBox
      data-id="text"
      display="block"
      component={component}
      color={color}
      className={className}
      textAlign={align}
    >
      {children}
    </StyledBox>
  );
};

Text.defaultValues = {
  color: "grey",
};

Text.displayName = "Text";
