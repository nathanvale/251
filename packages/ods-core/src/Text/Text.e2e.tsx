import React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Stack } from "../Stack";
import { Box } from "../Box";
import { Text } from "./Text";

export const tests: E2ETests = [
  {
    label: "Default",
    Code: () => <Text>The quick brown fox jumps over the lazy dog.</Text>,
    responsive: true,
  },
  {
    label: "Body Small Text",
    Code: () => (
      <Text variant="body-small">
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
    responsive: true,
  },
  {
    label: "Subtitle Text",
    Code: () => (
      <Text variant="subtitle">
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
  },
  {
    label: "Subtitle Small Text",
    Code: () => (
      <Text variant="subtitle-small">
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
  },
  {
    label: "Caption Text",
    Code: () => (
      <Text variant="caption">
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
  },
  {
    label: "Overline Text",
    Code: () => (
      <Text variant="overline-text">
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
  },
  {
    label: "Tone (positive)",
    Code: () => (
      <Text tone="positive">The quick brown fox jumps over the lazy dog.</Text>
    ),
  },
  {
    label: "Tone (critical)",
    Code: () => (
      <Text tone="critical">The quick brown fox jumps over the lazy dog.</Text>
    ),
  },
  {
    label: "Tone (promoteC)",
    Code: () => (
      <Text tone="promoteC">The quick brown fox jumps over the lazy dog.</Text>
    ),
  },
  {
    label: "Weight (medium)",
    Code: () => (
      <Text weight="medium">The quick brown fox jumps over the lazy dog.</Text>
    ),
  },
  {
    label: "Truncating long text",
    Code: () => (
      <Box style={{ width: 90 }}>
        <Text truncate>The quick brown fox jumps over the lazy dog.</Text>
      </Box>
    ),
  },
  {
    label: "Text Alignment",
    Code: () => (
      <Stack space="medium">
        <Text>The quick brown fox jumps over the lazy dog.</Text>
        <Text align="center">The quick brown fox jumps over the lazy dog.</Text>
        <Text align="right">The quick brown fox jumps over the lazy dog.</Text>
      </Stack>
    ),
  },
  {
    label: "Text Alignment (responsive)",
    Code: () => (
      <Text align={["left", "center"]}>
        The quick brown fox jumps over the lazy dog.
      </Text>
    ),
  },
  {
    label: "Inline Text",
    Code: () => (
      <Text>
        The quick brown fox{" "}
        <Text inline tone="positive">
          jumps over
        </Text>{" "}
        the lazy dog.
      </Text>
    ),
  },
];
