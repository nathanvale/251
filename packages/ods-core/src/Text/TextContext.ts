import { createContext } from "react";
import { UseTextProps } from "./Text";

export const TextContext = createContext<UseTextProps | false>(false);
