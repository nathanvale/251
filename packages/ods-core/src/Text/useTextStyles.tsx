import * as React from "react";
import clsx from "clsx";

import {
  useBasekickStyles,
  useStrongStyles,
  useToneStyles,
  useWeightStyles,
} from "@origin-digital/ods-typography";

import { useCheckTypographyBackground } from "../_private/hooks";
import { TextContext, UseTextProps } from "./TextContextProvider";

export function useTextStyles(props: UseTextProps) {
  const { tone = "neutral", weight, variant, inline } = props;
  useCheckTypographyBackground();
  const basekickStyles = useBasekickStyles({ variant, inline });
  const toneStyles = useToneStyles({ tone });
  const weightStyles = useWeightStyles(weight);
  const strongStyles = useStrongStyles("medium");
  return clsx(basekickStyles, toneStyles, weightStyles, strongStyles);
}

export function useTextStylesFromContext() {
  const textContext = React.useContext(TextContext);

  if (!textContext) {
    throw Error(
      [
        "The context is required for tone, variant and weight values. ",
        "You can only use this hook inside children of TextContextProvider. Make sure you component is wrapped in",
        "a TextContextProvider or one of 'Text' or 'List' components (which contain a TextContextProvider).",
      ].join(" ")
    );
  }

  const className = useTextStyles(textContext);

  return {
    className,
    ...textContext,
  };
}
