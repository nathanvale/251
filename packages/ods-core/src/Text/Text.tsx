import React, { ReactNode, useMemo } from "react";
import clsx from "clsx";
import { BoxProps, Box } from "@origin-digital/ods-core";
import {
  TextVariants,
  TextToneVariants,
  TypographyWeightVariants,
} from "@origin-digital/ods-types";
import {
  useBasekickStyles,
  useToneStyles,
  useWeightStyles,
  useStrongStyles,
} from "@origin-digital/ods-typography";
import {
  useCheckTypographyBackground,
  useTruncatedContent,
} from "../_private/hooks";

import { TextContext } from "./TextContext";

export interface UseTextProps {
  mediumWeight?: boolean;
  variant?: TextVariants;
  tone?: TextToneVariants;
}

export interface TextProps {
  children?: ReactNode;
  variant?: TextVariants;
  tone?: TextToneVariants;
  weight?: Exclude<TypographyWeightVariants, "bold">;
  align?: BoxProps["textAlign"];
  truncate?: boolean;
  "data-id"?: string;
  component?: BoxProps["component"];
}

interface UseTextStylesProps {
  weight: Exclude<TypographyWeightVariants, "bold">;
  tone?: TextToneVariants;
  variant: TextVariants;
}

const defaultWeight = "regular";
const defaultVariant = "body";

function useTextStyles(props: UseTextStylesProps) {
  const { tone = "neutral", weight, variant } = props;
  useCheckTypographyBackground();
  const basekickStyles = useBasekickStyles({ variant });
  const toneStyles = useToneStyles({ tone });
  const weightStyles = useWeightStyles(weight);
  const strongStyles = useStrongStyles("medium");
  return clsx(basekickStyles, toneStyles, weightStyles, strongStyles);
}

export const Text = (props: TextProps) => {
  const {
    children,
    component = "span",
    variant = defaultVariant,
    truncate = false,
    align,
    weight = defaultWeight,
    tone,
    "data-id": dataId,
  } = props;

  // Prevent re-renders when context values haven't changed
  const textContextValue = useMemo(
    () => ({
      tone,
      weight,
      variant,
    }),
    [tone, weight, variant]
  );

  const className = useTextStyles({ weight, tone, variant });

  const content = useTruncatedContent({
    children,
    truncate,
  });

  return (
    <TextContext.Provider value={textContextValue}>
      <Box
        data-id={dataId}
        className={className}
        component={component}
        textAlign={align}
      >
        {content}
      </Box>
    </TextContext.Provider>
  );
};

Text.defaultProps = {
  "data-id": "text",
  component: "span",
  variant: defaultVariant,
  tone: "neutral",
  weight: defaultWeight,
  align: "left",
  truncate: false,
};

Text.displayName = "Text";
