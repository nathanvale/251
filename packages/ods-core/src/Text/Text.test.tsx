import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Text } from "./Text";
import { Box } from "..";

test("It can render body text", () => {
  const { container } = render(<Text>Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render body small text", () => {
  const { container } = render(<Text variant="body-small">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render subtitle text", () => {
  const { container } = render(<Text variant="subtitle">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render subtitle small text", () => {
  const { container } = render(<Text variant="subtitle-small">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render subtitle caption text", () => {
  const { container } = render(<Text variant="caption">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render subtitle overline text", () => {
  const { container } = render(<Text variant="overline-text">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render body text medium weight", () => {
  const { container } = render(<Text weight="medium">Body Text</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render body text truncated", () => {
  const { container } = render(
    <Box style={{ width: 90 }}>
      <Text truncate>Truncated text</Text>
    </Box>
  );
  expect(container.firstChild).toMatchSnapshot();
});
