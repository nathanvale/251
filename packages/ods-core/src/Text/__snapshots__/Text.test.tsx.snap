// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can render body small text 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-small-variant-basekick-6 typography-neutral-color-7 typography-regular-weight-8 typography-strong-support-9 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text "critical" tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-36 typography-critical-color-37 typography-medium-weight-38 typography-strong-support-39 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text "neutral" tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-56 typography-neutral-color-57 typography-medium-weight-58 typography-strong-support-59 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text "neutral.dark" tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-46 typography-neutral.dark-color-47 typography-medium-weight-48 typography-strong-support-49 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text "neutral.light" tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-51 typography-neutral.light-color-52 typography-medium-weight-53 typography-strong-support-54 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text "positive" tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-41 typography-positive-color-42 typography-medium-weight-43 typography-strong-support-44 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text critical tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-61 typography-critical-color-62 typography-medium-weight-63 typography-strong-support-64 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text medium weight 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-31 typography-neutral-color-32 typography-medium-weight-33 typography-strong-support-34 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render body text truncated 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
  overflow: hidden;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<div
  class="c0"
  style="width: 90px;"
>
  <span
    class="typography-body-variant-basekick-66 typography-neutral-color-67 typography-regular-weight-68 typography-strong-support-69 -span c1"
    data-id="text"
  >
    <div
      class="typography-truncate-70 c2"
      display="block"
      overflow="hidden"
    >
      Truncated text
    </div>
  </span>
</div>
`;

exports[`It can render subtitle caption text 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-caption-variant-basekick-21 typography-neutral-color-22 typography-regular-weight-23 typography-strong-support-24 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render subtitle overline text 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-overline-text-variant-basekick-26 typography-neutral-color-27 typography-regular-weight-28 typography-strong-support-29 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render subtitle small text 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-subtitle-small-variant-basekick-16 typography-neutral-color-17 typography-regular-weight-18 typography-strong-support-19 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It can render subtitle text 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-subtitle-variant-basekick-11 typography-neutral-color-12 typography-regular-weight-13 typography-strong-support-14 -span c0"
  data-id="text"
>
  Body Text
</span>
`;

exports[`It should render a p tag by default 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<span
  class="typography-body-variant-basekick-1 typography-neutral-color-2 typography-regular-weight-3 typography-strong-support-4 -span c0"
  data-id="text"
>
  Body Text
</span>
`;
